intelmq/bots/parsers/alienvault/parser_otx.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""
+; Events are gathered based on user subscriptions in AlienVault OTX
+; The data structure is described in detail here:
+; https://github.com/AlienVault-Labs/OTX-Python-SDK/blob/master/howto_use_python_otx_api.ipynb
+"""
+from __future__ import unicode_literals
+import sys
+import json
+
+from intelmq.lib import utils
+from intelmq.lib.bot import Bot
+from intelmq.lib.harmonization import DateTime
+from intelmq.lib.message import Event
+
+HASHES = {
+    'FileHash-SHA256': '$5$',
+    'FileHash-SHA1': '$sha1$',
+    'FileHash-MD5': '$1$'
+}
+
+class AlienVaultOTXParserBot(Bot):
+
+    def process(self):
+        report = self.receive_message()
+        if (report is None or not report.contains("raw")):
+            self.acknowledge_message()
+            return
+
+        raw_report = utils.base64_decode(report.value("raw"))
+
+        for pulse in json.loads(raw_report):
+            additional = json.dumps(
+                {"author": pulse['author_name'], "pulse": pulse['name']})
+            for indicator in pulse["indicators"]:
+                event = Event()
+                # hashes
+                if indicator["type"] in HASHES.keys():
+                    event.add('malware.hash',
+                              HASHES[indicator["type"]] + indicator["indicator"])
+                # fqdn
+                if indicator["type"] in ['hostname', 'domain']:
+                    event.add('source.fqdn',
+                              indicator["indicator"], sanitize=True)
+                # IP addresses
+                elif indicator["type"] in ['IPv4', 'IPv6']:
+                    event.add('source.ip',
+                              indicator["indicator"], sanitize=True)
+                # emails
+                elif indicator["type"] == 'email':
+                    event.add('source.account',
+                              indicator["indicator"], sanitize=True)
+                # URLs
+                elif indicator["type"] in ['URL', 'URI']:
+                    event.add('source.url',
+                              indicator["indicator"], sanitize=True)
+                # CIDR
+                elif indicator["type"] in ['CIDR']:
+                    event.add('source.network',
+                              indicator["indicator"], sanitize=True)
+                # FilePath, Mutex, CVE - TODO: process these IoCs as well
+                else:
+                    continue
+
+                event.add('comment', pulse['description'])
+                event.add('additional', additional, sanitize=True)
+                event.add('classification.type', 'blacklist', sanitize=True)
+                event.add('time.observation', report.value(
```

Comment:
See the code in `lib/message.py`: https://github.com/certtools/intelmq/blob/master/intelmq/lib/message.py#L181
The Event-constructor differentiates between a Report and other parameter types like dict. Report is derived from Message which is in turn derived from dict. For a report instance, it only takes `feed.(name|url)` and `time.observation`, not `raw`. For everything else, like dicts and messages, the parameter is given to the constructor of the parent class, which is in the end dict, and all fields are copied. If this is really the case, it does mean that `receive_message` / the pipeline does not correctly unserialize the message!
So first please check the type of report e.g. `self.logger.debug('report type {!r}'.format(type(report)))`, and/or some `issubclass` calls.